pkg/descheduler/strategies/topologyspreadconstraint.go

```diff
@@ -76,7 +76,7 @@ func RemovePodsViolatingTopologySpreadConstraint(
 	nodes []*v1.Node,
 	podEvictor *evictions.PodEvictor,
 ) {
-	thresholdPriority, includedNamespaces, excludedNamespaces, err := validateAndParseTopologySpreadParams(ctx, client, strategy.Params)
+	includeSoftConstraints, thresholdPriority, includedNamespaces, excludedNamespaces, err := validateAndParseTopologySpreadParams(ctx, client, strategy.Params)
```

Comment:
I don't think `includeSoftConstraints` needs to be validated by this function, as it just complicates the function signature (and isn't actually validated, since it's just a bool).

We can just use `params.IncludeSoftConstraints` on L129 and add a nil check in that if statement for `params` so we don't get a dereference panic. wdyt?

Reply:
I originally wanted to check `params` only once so I put it in `validateAndParseTopologySpreadParams`.

It's more concise to do the nil check inside that if statement, I'll update soon.